opal/include/opal/sys/arm64/timer.h

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2008      The University of Tennessee and The University
+ *                         of Tennessee Research Foundation.  All rights
+ *                         reserved.
+ * Copyright (c) 2016      Broadcom Limited. All rights reserved.
+ * $COPYRIGHT$
+ *
+ * Additional copyrights may follow
+ *
+ * $HEADER$
+ */
+
+#ifndef OPAL_SYS_ARCH_TIMER_H
+#define OPAL_SYS_ARCH_TIMER_H 1
+
+#include <sys/times.h>
+
+typedef uint64_t opal_timer_t;
+
+static inline opal_timer_t
+opal_sys_timer_get_cycles(void)
+{
+    opal_timer_t ret;
+
```

Comment:
@shamisp  
You are right, reads can occur out of order and so it should be
isb(); asm volatile("mrs %0, cntvct_el0" : "=r" (ret));  
And yes, this timer increments at a fixed frequency.